bootstrap-app/src/cmr/bootstrap/services/dispatch/core_async_dispatch.clj

```diff
@@ -0,0 +1,143 @@
+(ns cmr.bootstrap.services.dispatch.core-async-dispatch
```

Comment:
I'm sure you are right. A couple of reasons I went with the approach I did:

1) Naming stuff with redundant names. We all tend to use atom/sublime type editors where only the filename is displayed in the tab. At a retrospective awhile back people asked that we include descriptive, though redundant based on the path, information rather than have 5 tabs open named core.clj. Probably a bad decision.
2) I treated the protocol definition just like I would an interface class in Java. The interface would not know anything about its implementors. I like a couple of things about this approach. The implementor clearly shows what protocol it is implementing, and it feels like the right place for calling extend directly in the namespace.

I think I like the core namespace idea being on the only public namespace, so I'll look to make the change in my next PR.